spec/lib/add_missing_asset_audit_records_spec.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+RSpec.describe 'asset_audit:add_missing_records', type: :task do
+  let(:file_path) { File.join('spec', 'data', 'asset_audits', 'data.csv') }
+
+  before do
+    Rake.application.rake_require 'tasks/add_missing_asset_audit_records'
+    Rake::Task.define_task(:environment)
+    allow(File).to receive(:exist?).and_return(true) # Stub default value
```

Comment:
Stubbing `File` like this is too broad and it will have side effects. It may even break the framework itself during the test. Please use fixture files instead.

Reply:
Ok, Mostly I have used mocking in similar situations in past where CSV.read is mocked to simulate specific conditions which I thought is very useful and easy to test different edge cases associated with parsing and validation. Thought mocks will give a higher level isolation, that tests do not depend on the filesystem.
But, I can see the value in your suggestion  as well of using different files for each case, which provides more realistic testing. I think fixture files also make the tests more readable and understandable. 
I will update my approach based on your suggestion.

Reply:
Ok, I have already removed it in the latest changes with the PR update.